libxrdp/xrdp_caps.c

```diff
@@ -539,6 +539,12 @@ xrdp_caps_process_frame_ack(struct xrdp_rdp *self, struct stream *s, int len)
     g_writeln("xrdp_caps_process_frame_ack:");
     self->client_info.use_frame_acks = 1;
     in_uint32_le(s, self->client_info.max_unacknowledged_frame_count);
+    if (self->client_info.max_unacknowledged_frame_count < 0)
+    {
+        g_writeln("  invalid max_unacknowledged_frame_count value (%d), setting to 0",
```

Comment:
I think we should avoid g_writeln() in the new and modified code. That would be `log_message(LOG_LEVEL_ERROR, ...)`

Also, is the largest positive integer a good limit? It could be a problem too if streams go out-of-sync. Let's have some reasonable limit. Maybe 8192.